wurst/math/Matrices.wurst

```diff
@@ -0,0 +1,310 @@
+package Matrices
+import NoWurst
+import Vectors
+import String
+
+
+/* Great thanks to euclideanspace.com for wide explanations. */
+
+
+public let ZERO22 = mat2(0, 0, 0, 0)
+public let ZERO33 = mat3(0, 0, 0, 0, 0, 0, 0, 0, 0)
+
+public let IDENTITY22 = mat2(
+    1, 0,
+    0, 1)
+
+public let IDENTITY33 = mat3(
+    1, 0, 0,
+    0, 1, 0,
+    0, 0, 1)
+
+
+/* =========== 2X2 MATRIX =========== */
+/** Matrxi 2x2
+00 01
+10 11
+*/
+public tuple mat2(real rc00, real rc01, real rc10, real rc11)
+
+                     
+public function mat2.op_plus(mat2 m) returns mat2
+    return mat2(
+        this.rc00 + m.rc00, this.rc01 + m.rc01,
+        this.rc10 + m.rc10, this.rc11 + m.rc11)
+
+public function mat2.op_minus(mat2 m) returns mat2
+    return mat2(
+        this.rc00 - m.rc00, this.rc01 - m.rc01,
+        this.rc10 - m.rc10, this.rc11 - m.rc11)
+
+public function mat2.op_minus(real scalar) returns mat2
+    return mat2(
+        this.rc00 - scalar, this.rc01 - scalar,
+        this.rc10 - scalar, this.rc11 - scalar)
+
+public function mat2.op_plus(real scalar) returns mat2
+    return mat2(
+        this.rc00 + scalar, this.rc01 + scalar,
+        this.rc10 + scalar, this.rc11 + scalar)
+    
+public function mat2.op_mult(real scalar) returns mat2
+    return mat2(
+        this.rc00*scalar, this.rc01*scalar,
+        this.rc10*scalar, this.rc11*scalar)
+
+public function real.op_mult(mat2 m) returns mat2
+    return m*this
+
+/** Matrix multiplication is not commutative that means matrix*vect is not the same as vec*matrix. */
+public function mat2.op_mult(vec2 v) returns vec2
+    return vec2(this.rc00*v.x + this.rc01*v.y, this.rc10*v.x + this.rc11*v.y)
+
+/** Matrix multiplication is not commutative that means matrix*vect is not the same as vec*matrix. */
+public function vec2.op_mult(mat2 m) returns vec2
+    return vec2(this.x*m.rc00 + this.y*m.rc10, this.x*m.rc01 + this.y*m.rc11)
+
+public function mat2.op_mult(mat2 m) returns mat2
+    return mat2(
+        this.rc00*m.rc00 + this.rc01*m.rc10, this.rc00*m.rc01 + this.rc01*m.rc11, 
+        this.rc10*m.rc00 + this.rc11*m.rc10, this.rc10*m.rc01 + this.rc11*m.rc11)
+
+/** Returns matrix' column by id as a vector or zero-vector if doesn't exist. */
+public function mat2.col(int id) returns vec2

```

Comment:
Can you test how well this (and .row()) get inlined?
Consider using a temp-var maybe instead of separate return statements.

Reply:
Didn't think about inlining at all. You're right, I'll rewrite it with a temporary variable.